vermouth/processors/generate_coordinates.py

```diff
@@ -0,0 +1,397 @@
+                    if px.max() == np.nan:
+                        continue
+                    # Split the limbs to new segments and return those
+                    #out = []
+                    for t, pi in zip(target, px):
+                        self.molecule.nodes[t]['position'] = pi
+                    #    for k in self.limbs:
+                    #        if len(k) > 1 and t in k:
+                    #            k.discard(t)
+                    #            for g in nx.connected_components(self.molecule.subgraph(k)):
+                    #                out.append(
+                    #                    Segment(
+                    #                        self.molecule,
+                    #                        [(t, [[a]], g)],
+                    #                        self.coords,
+                    #                        self.contact
+                    #                    )
+                    #                )
+                    #  This commented-out stuff may be useful for speeding up
+                    #  coordinate generation, because it avoids needing lookup
+                    #  of missing coordinates in subsequent cycles.
+                    #return out
+                    return
+                break
+        return None
+
+
+class GenerateCoordinates(Processor):
+    """
+    A processor for generating coordinates for atoms without.
+    """
+    def __init__(self, contact=0.5):
+        super().__init__()
+        self.contact = contact
+        self.coordinates = None
+
+    def run_system(self, system):
+        """
+        Process `system`.
+
+        Parameters
+        ----------
+        system: vermouth.system.System
+            The system to process. Is modified in-place.
+        """
+
+        self.coordinates = np.stack(
+            [
+                a.get('position', (np.nan, np.nan, np.nan))
+                for m in system.molecules
+                for i, a in m.atoms
+            ]
+        )
+
+        for molecule in system.molecules:
+            self.run_molecule(molecule)
+
+    def run_molecule(self, molecule):
+        """
+        Process a single molecule. Must be implemented by subclasses.
+
+        Parameters
+        ----------
+        molecule: vermouth.molecule.Molecule
+            The molecule to process.
+
+        Returns
+        -------
+        vermouth.molecule.Molecule
+            The provided molecule with complete coordinates
+        """
+
+        # Missing atoms - those without 'positions'
+        missing = get_missing_atoms(molecule)
+
+        prev = len(molecule)
+        while prev - len(missing) > 0:
+            # Limbs: subgraphs of missing atoms with anchors and bases
+            limbs = []
+            for g in nx.connected_components(molecule.subgraph(missing)):
+                anchors = list(get_anchors(molecule, g))
+                bases = [ list(get_anchors(molecule, [a])) for a in anchors ]
```

Comment:
`list` call unneeded

Reply:
get_anchors returns a set and there is indexing downstream, so I need lists of anchors. But it may be useful to have sets of anchors somewhere, so returning a set is proper and I just need to cast the type where I need it.

Reply:
Ah yes, I see now. It's indeed not a superfluous call.